### ISSUE TITLE
Review client creation methods

### ISSUE THREAD

**Issue by jskeet** [googleapis/gapic-generator-csharp#207]
We're currently generating what we used to generate, i.e.

- CreateAsync(gaxgrpc::ServiceEndpoint endpoint = null, XyzSettings settings = null)
- Create(gaxgrpc::ServiceEndpoint endpoint = null, XyzSettings settings = null)
- Create(grpccore::Channel channel, XyzSettings settings = null)
- Create(grpccore::CallInvoker callInvoker, XyzSettings settings = null)

As we're taking a breaking change anyway, and we now have the client builder pattern, we should check that we really want all of those.

**Comment by jskeet** [comment]
Initial thoughts on this... I'd like to remove the `Create(Channel, Settings)` call. Reasons for this:

- We'd almost certainly want this to change to be `ChannelBase` anyway
- When you've got a channel, it's easy to construct a `CallInvoker`
- We might want to remove the `endpoint` parameter from the methods that take that

Basically that leaves options for two scenarios:

- Create or CreateAsync where you want to use default credentials; the async version is required here because getting default credentials may involve IO
- Create accepting a CallInvoker for when you've got one already; this is what the ClientBuilder would use.

For anything more complicated, use the relevant client builder. We could potentially make the `Create(CallInvoker,Settings)` method internal, just to be used by the client builder (which allows the call invoker to be specified as a property).

Thoughts welcome.

**Comment by chrisdunelm** [comment]
I'd be happy with removing both the `Channel` and `CallInvoker` methods. Insisting on using the builder for anything complex seems reasonable to me.

**Comment by jskeet** [comment]
I'd forgotten that we can remove the CallInvoker method entirely as within the builder we can just call the ClientImpl constructor :)

**Comment by jskeet** [comment]
Note that I'd suggest the `Create` and `CreateAsync` methods can just delegate to the client builders, e.g. 

```csharp
public static FooClient Create(FooClientSettings settings = null) => new FooClientBuilder { Settings = settings }.Build();
public static FooClient CreateAsync(FooClientSettings settings = null, CancellationToken cancellationToken = default) =>
    new FooClientBuilder { Settings = settings }.BuildAsync(cancellationToken);
```

**Comment by jskeet** [comment]
I think I'll prototype this so we know what it would look like.

**Comment by chrisdunelm** [comment]
Yes, agreed on the `Create` methods delegating to the builder (although note your examples are missing the endpoint parameter!).

**Comment by jskeet** [comment]
Yup, that's deliberate - see an earlier bullet point. I suspect specifying the endpoint is incredibly rare. This is basically "helper methods for defaults other than Settings, builder for everything else." I could be persuaded to remove the settings too :)

**Comment by chrisdunelm** [comment]
ok, sgtm

**Comment by amanda-tarafa** [comment]
All sgtm, but I would remove Settings as well from the Create... methods. I would argue that specifying specific call settings for different methods is as rare and more "advanced" than specifying the endpoint?

**Comment by jskeet** [comment]
I suspect it's not quite as rare, but close enough. Will keep the factory methods super-simple.

**Comment by jskeet** [comment]
Having now implemented this, I really like the result. It should be very easy to write up in user guides etc: for simple situations, use the factory methods; for everything else, use the client builder.

**Comment by jskeet** [comment]
This was implemented in #212
